exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/SpecialFunctions.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.expr.fn.impl;
+
+import org.apache.drill.exec.expr.DrillSimpleFunc;
+import org.apache.drill.exec.expr.annotations.FunctionTemplate;
+import org.apache.drill.exec.expr.annotations.Output;
+import org.apache.drill.exec.expr.annotations.Param;
+import org.apache.drill.exec.vector.complex.reader.FieldReader;
+import org.apache.drill.exec.vector.complex.writer.BaseWriter;
+
+/**
+ * Container class for functions with {@link org.apache.calcite.sql.SqlSyntax#SPECIAL} syntax.
+ */
+public class SpecialFunctions {
+
+  private SpecialFunctions() {
+  }
+
+  /**
+   * Implementation of ROW(col1, col2, ..., colN) constructor function. Most often it's
+   * used to reconstruct struct columns after Calcite's flattening.
+   */
+  @FunctionTemplate(name = "row", scope = FunctionTemplate.FunctionScope.SIMPLE, isVarArg = true)
```

Comment:
It would be useful to point out that this function expects arguments (field name and field value), ...